### PR TITLE
Fix: Issue#945  Sunspot::UnrecognizedFieldError: (#1)

### PR DESCRIPTION
**Pull request by mgadikota-rpx**
related to issue: https://github.com/sunspot/sunspot/issues/945

### CODE REVIEW

**Comment by serggl** [comment]
@mgadikota-rpx this should include tests (sample code in the linked issue is the good candidate to be tested out)

**Comment by mgadikota-rpx** [comment]
@serggl  Will do.

**Comment by mgadikota-rpx** [comment]
@serggl  Just added two rpsec test cases, please review.

File: sunspot/spec/integration/field_lists_spec.rb [hunk 1]
```diff
@@ -31,6 +31,22 @@
     end
   end
 
+  it 'when listing existing text fields it should not raise Sunspot::UnrecognizedFieldError' do
```
Comment (serggl) [inline sunspot/spec/integration/field_lists_spec.rb]: can we change title wording to "does not raise ...." ?

File: sunspot/spec/integration/field_lists_spec.rb [hunk 2]
```diff
@@ -31,6 +31,22 @@
...
+    expect do
+      Sunspot.search(Post) {
+        field_list(:body) 
+      }
+    end.to_not raise_error
+  end
+
+  it 'when listing non-existing text fields it should raise Sunspot::UnrecognizedFieldError' do
```
Comment (serggl) [inline sunspot/spec/integration/field_lists_spec.rb]: same as above comment

File: sunspot/spec/integration/field_lists_spec.rb (hunk 1 above)
Comment (mgadikota-rpx) [reply sunspot/spec/integration/field_lists_spec.rb]: @serggl  Yes, definetely will do.

File: sunspot/spec/integration/field_lists_spec.rb (hunk 2 above)
Comment (mgadikota-rpx) [reply sunspot/spec/integration/field_lists_spec.rb]: @serggl  will do.

Comment (mgadikota-rpx) [reply sunspot/spec/integration/field_lists_spec.rb]: @serggl  done.

File: sunspot/spec/integration/field_lists_spec.rb (hunk 1 above)
Comment (mgadikota-rpx) [reply sunspot/spec/integration/field_lists_spec.rb]: @serggl  done.

File: sunspot/spec/integration/field_lists_spec.rb [hunk 3]
```diff
@@ -31,6 +31,22 @@
     end
   end
 
+  it 'when listing existing text fields it does not raise Sunspot::UnrecognizedFieldError' do
```
Comment (serggl) [inline sunspot/spec/integration/field_lists_spec.rb]: @mgadikota-rpx sorry to bother you :) but I was ment to say it should read like:
> does not raise Sunspot::UnrecognizedFieldError when listing existing text fields

Comment (mgadikota-rpx) [reply sunspot/spec/integration/field_lists_spec.rb]: @serggl  just made this change.

**Comment by mgadikota-rpx** [comment]
@serggl  thanks for your time though..

**Comment by mgadikota-rpx** [comment]
@serggl  Could we expect new version release ?

**Comment by serggl** [comment]
I will release this later this week to package along with some other PRs

**Pull request merged**